tools/bin/commands/util/openshift_funcs

```diff
@@ -222,3 +222,42 @@ guess_route() {
     fi
     echo "${project}.${domain#*.}"
 }
+
+create_api_cookie() {
+
+    local COOKIE1="$HOME/syndesis_cookie.txt"
+    # second cookie needed because curl flushes old values upon redirect. Might be a bug or some obscure flag missing
+    local COOKIE2="$(mktemp /tmp/syndesis_cookie.XXX.txt)"
+    # generate a new user so that it will be asked to approve permissions delegation
+    local USERNAME="dev$RANDOM"
+
+    # clean cookie
+    rm -rf "$COOKIE1"
+    touch "$COOKIE1"
+
+    local SYNDESIS_URL="$(oc get route syndesis  --template={{.spec.host}})"
+    # first invocation is needed just to get the redierect url and to define a csrf token
+    local REDIRECTED_URL="$(curl --write-out "%{url_effective}\n" --location --silent --show-error --output /dev/null --insecure --cookie-jar  "$COOKIE1" -v "https://$SYNDESIS_URL" 2>/dev/null )"
+
+    # isolate required params
+    local LOGIN_URL="$(echo $REDIRECTED_URL | sed -e 's/?/\n/g' | head -n1)"
+    local THEN_URL="$(echo $REDIRECTED_URL | sed -e 's/=/\n/g' | tail -n1)"
+    local CSRF=$(cat "$COOKIE1"  | grep -Po '(?<=\scsrf).*' | awk '{$1=$1;print}')
+
+    # login on OpenShift and save output to TEMP_STATE location
+    local TEMP_STATE="$(mktemp /tmp/syndesis_state.XXX.html)"
+    curl --referer "$REDIRECTED_URL" --insecure --location  -H "Cookie: csrf=$CSRF" --data-urlencode username="$USERNAME"  --data-urlencode password="developer" --data-ascii then="$THEN_URL" --data-urlencode csrf="$CSRF" --cookie-jar "$COOKIE2" -i -v -o "$TEMP_STATE" "https://$(minishift ip):8443/login"  2>/dev/null
```

Comment:
Not sure why use use `minishift ip` when you have $SYNDESIS_URL

Reply:
syndesis url is, syndesis url :wink:  

it includes the defined route.

`minishift ip` just gives you the ip. they are 2 different endpoints.